### PR TITLE
Added ability to install MySQL 8 

### PR DESCRIPTION
**Pull request by niktest**
This PR contains an implementation of the new installation switch `--mysql-classic`, which allows proceeding with the installation of the classic MySQL. 
Also, additionally refactored `v-add-sys-roundcube` and added support the MySQL/PostgreSQL.

### CODE REVIEW

**Comment by jaapmarcus** [comment]
👍🏼 I assume it installs Mysql 8?

**Comment by niktest** [comment]
> 👍🏼 I assume it installs Mysql 8?

Hi @jaapmarcus,
Yes, it's installing the most recent MySQL 8. I have tested this installation on the clean Ubuntu 20.04.5 LTS (GNU/Linux 5.15.0-1023-aws x86_64) and on clean Ubuntu 20.04.5 LTS (GNU/Linux 5.4.0-132-generic x86_64).
Unfortunately, I didn't have a chance to test it on Debian, but I have modified the script accordingly.
Ideally, it should be working the same.

**Comment by jaapmarcus** [comment]
Will set up a test server on Debian to verify if it works. 

Last time I tried I only had major issues storing the password hashes Mysql8 generated when you run v-add-database

**Comment by niktest** [comment]
@jaapmarcus Honestly, I'm also probably faced the same issue couple of times and I thought it was just random staff. It happens from time to time.
I have created 4 WordPress installations, and have a problem with the visiting `/list/db/` page on some of them (not all). So I thought maybe it was an accidental issue.
The problem was with the incorrect special symbols which are appearing in the `/usr/local/hestia/data/users/$USERNAME/db.conf`, so when the page was calling `v-list-databases` it's causing the following error: `Error: Invalid object format: `

The problem is `MD5` field is contains `"` symbol in `db.conf`, which is causing the parsing error and then, throws the exception in `parse_object_kv_list`

**Comment by jaapmarcus** [comment]
> @jaapmarcus Honestly, I'm also probably faced the same issue couple of times and I thought it was just random staff. It happens from time to time. I have created 4 WordPress installations, and have a problem with the visiting `/list/db/` page on some of them (not all). So I thought maybe it was an accidental issue. The problem was with the incorrect special symbols which are appearing in the `/usr/local/hestia/data/users/$USERNAME/db.conf`, so when the page was calling `v-list-databases` it's causing the following error: `Error: Invalid object format: `
> 
> The problem is `MD5` field is contains `"` symbol in `db.conf`, which is causing the parsing error and then, throws the exception in `parse_object_kv_list`

See https://github.com/hestiacp/hestiacp/issues/2622

Would be nice if we can solve this issue

**Comment by jaapmarcus** [comment]
I know MyVesta (Other fork of Vesta) resolved it by "forcing" mysql8 to use mysql5 legacy mode

https://github.com/myvesta/vesta/blob/e134202224757c32f97fb817ba03ee5105104432/install/vst-install-debian.sh#L758-L760

**Comment by niktest** [comment]
@jaapmarcus Yep, it looks like really related to this issue. Maye this mysql5 legacy mode is a better workaround as probably changes the logic of `parse_object_kv_list`  can cause a widespread of potential issues.
Another way is to escape the special characters before writing to `db.conf`

**Comment by niktest** [comment]
@jaapmarcus I have reproduced the issue. It happens with the ratio 1:5. I possibly found another workaround.
https://dev.mysql.com/doc/refman/8.0/en/server-system-variables.html#sysvar_print_identified_with_as_hex 

So, if we proceed with the MD5 creation with this command line, it's didn't cause the special symbols issue
`mysql -e "SET print_identified_with_as_hex=ON; SHOW CREATE USER olnik_tst;"`

I will try to play with this...

File: install/hst-install-debian.sh [hunk 1]
```diff
@@ -555,6 +559,9 @@ echo
 if [ "$mysql" = 'yes' ]; then
     echo '   - MariaDB Database Server'
 fi
+if [ "$mysqlclassic" = 'yes' ]; then
+    echo '   - MySQL Database Server'
+fi
```
Comment (jaapmarcus) [inline install/hst-install-debian.sh:567]: We should set $mysql to no when mysql classic is set to yes

**Review by jaapmarcus — COMMENTED** [review]
Also I get:

E: Package 'mysql-client' has no installation candidate
E: Package 'mysql-server' has no installation candidate

it looks like Debian doesn't have Mysql8 on default

**Comment by niktest** [comment]
> Also I get:
> 
> E: Package 'mysql-client' has no installation candidate E: Package 'mysql-server' has no installation candidate
> 
> it looks like Debian doesn't have Mysql8 on default

@jaapmarcus Thank you for checking. Seems it was too optimistic to think that it will be fine on Debian :-)
I need time to find a way how to install it on Debian...

File: install/hst-install-debian.sh (hunk 1 above)
Comment (jaapmarcus) [reply install/hst-install-debian.sh:567]: Also should be changed for Ubuntu

**Comment by jaapmarcus** [comment]
> 

If you contact me on Discord I am happy to give you access to a test vps. 

So it is only the time to test it ... 

https://github.com/myvesta/vesta/blob/master/install/vst-install-debian.sh#L729-L755
MyVesta does it like this so it should work fine

File: install/hst-install-debian.sh (hunk 1 above)
Comment (niktest) [reply install/hst-install-debian.sh:567]: I have pushed the fix and placed it in `# Checking software conflicts` section, if you don't mind

Comment (jaapmarcus) [reply install/hst-install-debian.sh:567]: 👍🏼

**Comment by niktest** [comment]
> If you contact me on Discord I am happy to give you access to a test vps.
> So it is only the time to test it ...
> https://github.com/myvesta/vesta/blob/master/install/vst-install-debian.sh#L729-L755 MyVesta does it like this so it should work fine

@jaapmarcus Thank you for the offer. But I'm worry that I may break something on your VPS and it will require a reset. Anyway happy to reach out to you directly, but I don't know how to find you. Here is my discord name `niktest#4151` 🙂

Tomorrow I will try to install and test on the following local VMWare images:
- debian-11.5.0-amd64-netinst.iso
- debian-10.12.0-amd64-netinst.iso

**Comment by jaapmarcus** [comment]
https://discord.gg/nXRUZch

Hestia discord server and the then jaapmarcus#0479

I have some vps servers that I only use for testing things like this...

But this will also work...

**Comment by jaapmarcus** [comment]
Installing on Ubuntu 22.04 works fine

```
You have successfully installed Hestia Control Panel on your server.

Ready to get started? Log in using the following credentials:

    Admin URL:  https://xxxxx:8083
    Username:   admin
    Password:   xxxxx
```

**Comment by jaapmarcus** [comment]
https://github.com/jaapmarcus/hestiacp/tree/test-mysql8

Should fix Debian 11 and 10 Feel free to merge it

**Comment by jaapmarcus** [comment]
I ran some "Standard" tests:

```
 ✓ MYSQL: Add database 
 ✓ MYSQL: Add Database (Duplicate) 
 ✗ MYSQL: Rebuild Database
   (from function `assert_success' in file test_helper/bats-assert/src/assert_success.bash, line 42,
    in test file test.bats, line 1681)
     `assert_success' failed with status 2
   
   -- command failed --
   status : 2
   output : Error: Invalid object format: DB='test-5285_database' DBUSER='test-5285_dbuser' MD5='\$A\$005\$^?6(.&:9N`CxK 6pynu5ANaDL03kv4HKDjsausbDmRho7aGlGkdSyc5R6' HOST='localhost' TYPE='mysql' CHARSET='UTF8' U_DISK='0' SUSPENDED='no' TIME='18:50:50' DATE='2022-11-22'
   --
   
 ✗ MYSQL: Change database user password
   (from function `assert_success' in file test_helper/bats-assert/src/assert_success.bash, line 42,
    in test file test.bats, line 1687)
     `assert_success' failed with status 2
   
   -- command failed --
   status : 2
   output : Error: Invalid object format: DB='test-5285_database' DBUSER='test-5285_dbuser' MD5='\$A\$005\$^?6(.&:9N`CxK 6pynu5ANaDL03kv4HKDjsausbDmRho7aGlGkdSyc5R6' HOST='localhost' TYPE='mysql' CHARSET='UTF8' U_DISK='0' SUSPENDED='no' TIME='18:50:50' DATE='2022-11-22'
   --
   
 ✗ MYSQL: Change database user
   (from function `assert_success' in file test_helper/bats-assert/src/assert_success.bash, line 42,
    in test file test.bats, line 1695)
     `assert_success' failed with status 2
   
   -- command failed --
   status : 2
   output : Error: Invalid object format: DB='test-5285_database' DBUSER='test-5285_dbuser' MD5='\$A\$005\$^?6(.&:9N`CxK 6pynu5ANaDL03kv4HKDjsausbDmRho7aGlGkdSyc5R6' HOST='localhost' TYPE='mysql' CHARSET='UTF8' U_DISK='0' SUSPENDED='no' TIME='18:50:50' DATE='2022-11-22'
   --
   
 ✗ MYSQL: Suspend database
   (from function `assert_success' in file test_helper/bats-assert/src/assert_success.bash, line 42,
    in test file test.bats, line 1702)
     `assert_success' failed with status 2
   
   -- command failed --
   status : 2
   output : Error: Invalid object format: DB='test-5285_database' DBUSER='test-5285_dbuser' MD5='\$A\$005\$^?6(.&:9N`CxK 6pynu5ANaDL03kv4HKDjsausbDmRho7aGlGkdSyc5R6' HOST='localhost' TYPE='mysql' CHARSET='UTF8' U_DISK='0' SUSPENDED='no' TIME='18:50:50' DATE='2022-11-22'
   --
   
 ✗ MYSQL: Unsuspend database
   (from function `assert_success' in file test_helper/bats-assert/src/assert_success.bash, line 42,
    in test file test.bats, line 1708)
     `assert_success' failed with status 6
   
   -- command failed --
   status : 6
   output : Error: db test-5285_database is not suspended
   --
   
 ✗ MYSQL: Delete database
   (from function `assert_success' in file test_helper/bats-assert/src/assert_success.bash, line 42,
    in test file test.bats, line 1714)
     `assert_success' failed with status 2
   
   -- command failed --
   status : 2
   output : Error: Invalid object format: DB='test-5285_database' DBUSER='test-5285_dbuser' MD5='\$A\$005\$^?6(.&:9N`CxK 6pynu5ANaDL03kv4HKDjsausbDmRho7aGlGkdSyc5R6' HOST='localhost' TYPE='mysql' CHARSET='UTF8' U_DISK='0' SUSPENDED='no' TIME='18:50:50' DATE='2022-11-22'
   --
   
 ✗ MYSQL: Delete missing database
   (from function `assert_failure' in file test_helper/bats-assert/src/assert_failure.bash, line 76,
    in test file test.bats, line 1720)
     `assert_failure $E_NOTEXIST' failed with status 2
   
   -- command failed as expected, but status differs --
   expected : 3
   actual   : 2
   output   : Error: Invalid object format: DB='test-5285_database' DBUSER='test-5285_dbuser' MD5='\$A\$005\$^?6(.&:9N`CxK 6pynu5ANaDL03kv4HKDjsausbDmRho7aGlGkdSyc5R6' HOST='localhost' TYPE='mysql' CHARSET='UTF8' U_DISK='0' SUSPENDED='no' TIME='18:50:50' DATE='2022-11-22'
   --
   
   ```

**Comment by jaapmarcus** [comment]
And:
```
 ✗ Backup: List Backups
   (from function `assert_success' in file test_helper/bats-assert/src/assert_success.bash, line 42,
    in test file test.bats, line 1999)
     `assert_success' failed with status 2
   
   -- command failed --
   status : 2
   output : Error: Invalid object format: BACKUP='test-5285.2022-11-22_18-52-21.tar' TYPE='local' SIZE='1' WEB='test-5285.hestiacp.com' DNS='test-5285.hestiacp.com' MAIL='test-5285.hestiacp.com' DB='Error:,Invalid,object,format:,DB='test-5285_database',DBUSER='test-5285_dbuser',MD5='\\$A\\$005\\$^?6(.&:9N`CxK,6pynu5ANaDL03kv4HKDjsausbDmRho7aGlGkdSyc5R6',HOST='localhost',TYPE='mysql',CHARSET='UTF8',U_DISK='0',SUSPENDED='no',TIME='18:50:50',DATE='2022-11-22'' CRON='' UDIR='.bash_logout,.bashrc,.cache,.cloud-locale-test.skip,.composer,.config,.local,.npm,.profile,.ssh,.vscode-server' RUNTIME='1' TIME='18:52:21' DATE='2022-11-22'
   --
   
 ✗ Backup: Delete backups
   (from function `assert_success' in file test_helper/bats-assert/src/assert_success.bash, line 42,
    in test file test.bats, line 2005)
     `assert_success' failed with status 2
   
   -- command failed --
   status : 2
   output : Error: invalid backup format :: Error:
   --
   
 ✓ Change: Change domain owner 
 ✗ Change: Add database
   (from function `assert_success' in file test_helper/bats-assert/src/assert_success.bash, line 42,
    in test file test.bats, line 2024)
     `assert_success' failed with status 4
   
   -- command failed --
   status : 4
   output : Error: DB=test-5285_database already exists
   --
   
 ✗ Change: Change database owner
   (from function `assert_success' in file test_helper/bats-assert/src/assert_success.bash, line 42,
    in test file test.bats, line 2032)
     `assert_success' failed with status 2
   
   -- command failed --
   status : 2
   output : Error: Invalid object format: DB='test-5285_database' DBUSER='test-5285_dbuser' MD5='\$A\$005\$^?6(.&:9N`CxK 6pynu5ANaDL03kv4HKDjsausbDmRho7aGlGkdSyc5R6' HOST='localhost' TYPE='mysql' CHARSET='UTF8' U_DISK='0' SUSPENDED='no' TIME='18:50:50' DATE='2022-11-22'
   --
   
 ✗ Change: Delete database
   (from function `assert_success' in file test_helper/bats-assert/src/assert_success.bash, line 42,
    in test file test.bats, line 2038)
     `assert_success' failed with status 3
   
   -- command failed --
   status : 3
   output : Error: db db test-5286_database doesn't exist
   --
   
 ✓ Mail: Delete domain 
 ✓ DNS: Delete domain 
 ✓ WEB: Delete domain 
 ✗ Delete user
   (from function `refute_output' in file test_helper/bats-assert/src/refute_output.bash, line 173,
    in test file test.bats, line 2067)
     `refute_output' failed
   
   -- output non-empty, but expected no output --
   output (15 lines):
     Error: invalid database format :: Error:
     Error: db db Invalid doesn't exist
     Error: db db object doesn't exist
     Error: invalid database format :: format:
     Error: invalid database format :: DB='test-5285_database'
     Error: invalid database format :: DBUSER='test-5285_dbuser'
     Error: invalid database format :: MD5='\$A\$005\$^?6(.&:9N`CxK
     Error: invalid database format :: 6pynu5ANaDL03kv4HKDjsausbDmRho7aGlGkdSyc5R6'
     Error: invalid database format :: HOST='localhost'
     Error: invalid database format :: TYPE='mysql'
     Error: invalid database format :: CHARSET='UTF8'
     Error: invalid database format :: U_DISK='0'
     Error: invalid database format :: SUSPENDED='no'
     Error: invalid database format :: TIME='18:50:50'
     Error: invalid database format :: DATE='2022-11-22'
   --
   
 ✓ Delete user2 
 ```

**Comment by jaapmarcus** [comment]
I pushed the small change for Debian support to the branch.

**Comment by niktest** [comment]
Hi @jaapmarcus,
I've pushed the changes that should potentially fix the tests.
Is there any chance that you may re-run the tests, please?

**Review by jaapmarcus — APPROVED** [review]

**Review by ScIT-Raphael — APPROVED** [review]

**Pull request merged**